### PR TITLE
Updated logic

### PR DESCRIPTION
**Pull request by ryanmitchell**
My branch seemed to be out of date, so wasnt using the same logic as the latest.
This is better.

### CODE REVIEW

File: components/LocalBox.php [hunk 1]
```diff
@@ -243,16 +243,14 @@ protected function updateCurrentOrderType()
...
+            if (!count($locationHasOrderTypes))
+                return;
 
-        // make sure location supports default
-        if (!in_array($defaultOrderType, $locationHasOrderTypes))
             $defaultOrderType = array_shift($locationHasOrderTypes);
+        }
 
-        $this->location->updateOrderType($defaultOrderType);
+        if (!$sessionOrder = $this->location->getSession('orderType') OR !in_array($sessionOrder, $locationCurrent->availableOrderTypes()))
```
Comment (sampoyigi) [inline components/LocalBox.php:253]: Rename `$locationHasOrderTypes` to `$locationOrderTypes` and use it here..

**Review by sampoyigi — CHANGES_REQUESTED** [review]

**Comment by ryanmitchell** [comment]
I can’t use it as it may have been array_shifted... 

I could change the shift to a [0] I suppose...

**Comment by sampoyigi** [comment]
Yea change to [0]

**Comment by ryanmitchell** [comment]
Done

**Pull request merged**